nifi-nar-bundles/nifi-jmx-bundle/nifi-jmx-processors/src/main/java/org/apache/nifi/processors/jmx/GetJMX.java

```diff
@@ -0,0 +1,385 @@
+            for (ObjectName mbean : mbeans) {
+                String domain = mbean.getDomain();
+
+                Record record = new Record(jsonBuilderFactory);
+                Hashtable<String, String> mbeanProperties = mbean.getKeyPropertyList();
+
+                if (mbeanProperties.containsKey("type") && canProcess(domain, mbeanProperties.get("type"))) {
+                    record.setDomain( domain );
+                    record.setProperties(mbeanProperties.entrySet());
+
+                    MBeanInfo mbeanInfo = mbsc.getMBeanInfo(mbean);
+                    MBeanAttributeInfo[] mbeanAttributeInfoAr = mbeanInfo.getAttributes();
+
+                    boolean validRecord = true;
+
+                    Map<String, String> mbeanStringMap = new HashMap<String, String>();
+
+                    for (MBeanAttributeInfo mbeanAttributeInfo : mbeanAttributeInfoAr) {
+                        try {
+                            Object value = mbsc.getAttribute(mbean, mbeanAttributeInfo.getName());
+
+                            mbeanStringMap.put( mbeanAttributeInfo.getName(), value.toString() );
+                        } catch (Exception e) {
+                            // IF ERROR DO NOT ADD to metricsSet
+                            validRecord = false;
+                            getLogger().warn("Exception fetching MBean attribute: " + e.getMessage() +
+                                    ": details: [" + mbean.getCanonicalName() + "]");
+                        }
+                    }
+
+                    if (validRecord) {
+                        record.setAttributes(mbeanStringMap);
+                        metricsSet.add(record);
+                    }
+                } else {
+                    getLogger().info("SKIPPING: domain [" + domain + "] type [" + mbeanProperties.get("type") + "]");
+                }
+            }
+
+            jmxc.close();
+        } catch( ProcessException pe ) {
+            throw pe;
+        } catch( IOException ioe) {
+            getLogger().error( "Exception connecting to JMX RMI Listener: " + ioe.getMessage() + ": hostname [" +
+                    hostname + "] port [" + port + "]");
+        } catch( SecurityException se ) {
+          getLogger().error( "Exception connecting to JMX RMI Listener due to security issues: " +
+                    se.getMessage() + ": hostname [" + hostname + "] port [" + port + "]" );
+        } catch( InstanceNotFoundException|IntrospectionException|ReflectionException e ) {
+            getLogger().error( "Exception with MBean Server: " + e.getMessage() );
+        } catch( Exception e ) {
+            getLogger().error( "Exception performing MBean Query: " + e.getMessage() );
+        }
+
+        return metricsSet;
+    }
+
+
+
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+        if (queue.size() < 100) {
```

Comment:
I see. I would still avoid hardcoded values.

Also, from what I understand the 100 will interact with the `BATCH_SIZE `property. Haven't tested but I suspect a `BATCH_SIZE `larger than 100 would be bogus?